src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -487,7 +488,8 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
             {
                 object? bodyValue = defaultBodyValue;
 
-                if (httpContext.Request.ContentLength != 0 && httpContext.Request.HasJsonContentType())
+                var feature = httpContext.Features.Get<IHttpRequestBodyDetectionFeature>();
```

Comment:
How cheap is this check @halter73 @Tratcher ?

Reply:
Very. Kestrel usually does the check once at the start of a request and stores the result. This property only exposes existing server knowledge.